integration_tests/__tests__/start.test.js

```diff
@@ -57,14 +34,48 @@ function testPlatform(platform, done) {
 
     if (message.match(/ERROR/g)) {
       done.fail(message);
-      haul.kill();
     }
 
-    if (message.match('bundling your React Native app')) {
-      const stdout = messageBuffer.join('\n');
-      expect(stdout).toMatch('INFO  Ready at http://localhost:8081');
+    if (message.match('running')) {
       done();
-      haul.kill();
     }
   });
+});
+beforeEach(() => cleanup(TEMP_DIR));
+afterEach(() => cleanup(TEMP_DIR));
+afterAll(() => {
+  haul.kill();
+});
+
+test('starts server and bundling iOS platform', done => {
+  testPlatform('ios', done);
+});
+
+test('starts server and bundling Android platform', done => {
+  testPlatform('android', done);
+});
+
+test('starts server and bundling all platforms', done => {
+  let calledTimes = 0;
+  function _done() {
+    calledTimes++;
+    if (calledTimes === 2) {
+      done();
+    }
+  }
+  _done.fail = done.fail;
+
+  testPlatform('ios', _done);
+  testPlatform('android', _done);
+});
+
+async function testPlatform(platform, done) {
```

Comment:
what do you think about changing this to return a promise instead of accepting a callback? I think tests will become simpler that way.

```js
async function testPlatform(platform) {
  const res = await fetch(`http://localhost:8081/index.${platform}.bundle`);
  const bundle = await res.text();
  expect(bundle).toMatch('__webpack_require__');
}
```

Then tests will be like:

```js
test('starts server and bundling iOS platform', () =>
  testPlatform('ios');
);
 
test('starts server and bundling Android platform', () =>
  testPlatform('android');
);
 
test('starts server and bundling all platforms', () =>
  Promise.all([
    testPlatform('ios');
    testPlatform('android');
  ]);
);
```

Reply:
With current on-demand bundling, does `start --platform` even make sense? I think we should remove or adjust these tests

Reply:
Test __doesn't__ spawn `haul start` with `--platform` option. `testPlatform` is only to make a request for specific platform bundle.

Reply:
@zamotany obviously, looks like I missed that 🤦‍♂️